test/Passes/uKernels/avx2/pass-vector-contract-to-FMAs.mlir

```diff
@@ -0,0 +1,133 @@
+// RUN: tpp-opt %s --tile-brgemm-linalg="registerBlocking=3,32,1" --loop-invariant-code-motion --vectorization-pass --hoist-vector-transfer --vector-contract-to-ukernels  --split-input-file  | FileCheck -check-prefix=CHECK %s
+// RUN: tpp-opt %s --tile-brgemm-linalg="registerBlocking=3,32,4" --loop-invariant-code-motion --vectorization-pass --hoist-vector-transfer --vector-contract-to-ukernels  --split-input-file  | FileCheck -check-prefix=CHECK1 %s
+// RUN: tpp-opt %s --tile-brgemm-linalg="registerBlocking=3,4,1" --loop-invariant-code-motion --vectorization-pass --hoist-vector-transfer --vector-contract-to-ukernels  --split-input-file  | FileCheck -check-prefix=CHECK2 %s
+// RUN: tpp-opt %s --tile-brgemm-linalg="registerBlocking=3,32,2" --loop-invariant-code-motion --vectorization-pass --hoist-vector-transfer --vector-contract-to-ukernels  --split-input-file  | FileCheck -check-prefix=CHECKBF16 %s
+// RUN: tpp-opt %s --tile-brgemm-linalg="registerBlocking=3,32,4" --loop-invariant-code-motion --vectorization-pass --hoist-vector-transfer --vector-contract-to-ukernels  --split-input-file  | FileCheck -check-prefix=CHECKBF161 %s
+// RUN: tpp-opt %s --tile-brgemm-linalg="registerBlocking=3,4,2" --loop-invariant-code-motion --vectorization-pass --hoist-vector-transfer --vector-contract-to-ukernels  --split-input-file  | FileCheck -check-prefix=CHECKBF162 %s
+
+module {
+ func.func @optimal_register_allocation(%arg0: memref<32x24x32xf32>, %arg1: memref<32x32x64xf32>, %arg2: memref<24x64xf32>) {
+     linalg.batch_reduce_matmul ins(%arg0, %arg1 : memref<32x24x32xf32>, memref<32x32x64xf32>) outs(%arg2 : memref<24x64xf32>)
+   return 
+ }
+}
+
+// CHECK-LABEL:   func.func @optimal_register_allocation(
```

Comment:
Do we really need super verbose testing like that?
Integration test already ensures correctness. This breaks and becomes unmaintainable after any minor change.
I'd suggest focusing on a small subset e.g., a few loads, their fmas, and writes.